### PR TITLE
macOS updates and fixes

### PR DESCRIPTION
**Pull request by vampirecat35**
Fixes and updates for macOS:
- Deprecations fixed
- LibreSSL added (with Apple specific patches)
- Fixed LibreSSL check
- Bugfixes
- Sandbox implemented for eID Viewer

### CODE REVIEW

**Comment by vampirecat35** [comment]
If you wish to test the eID Viewer / Middleware

Download:
https://github.com/vampirecat35/eid-mw/releases/tag/5.0.14

this contains the full build (for Big Sur).

File: cardcomm/ctkToken/BEIDToken.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -419,48 +422,35 @@
...
 			};
 			name = Debug;
 		};
 		2F3883AB1CF46C6F00F114A7 /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CODE_SIGN_IDENTITY = "Mac Developer";
 				COMBINE_HIDPI_IMAGES = YES;
-				DEVELOPMENT_TEAM = EU27N85PBZ;
```
Comment (yoe) [inline cardcomm/ctkToken/BEIDToken.xcodeproj/project.pbxproj:440]: No.

File: cardcomm/ctkToken/BEIDToken/TokenSession.m [hunk 2]
```diff
@@ -661,7 +661,7 @@ - (NSData *)generalAuthenticateWithData:(NSData *)data session:(BEIDTokenSession
         SecAsn1CoderCreate(&coder);
     
         SecAsn1Item dest = {0, nil};
-        OSStatus ortn = SecAsn1EncodeItem(coder, &ecdsa, ECDSATemplate, &dest);
+        /* OSStatus ortn = */ SecAsn1EncodeItem(coder, &ecdsa, ECDSATemplate, &dest);
```
Comment (yoe) [inline cardcomm/ctkToken/BEIDToken/TokenSession.m:664]: Please explain in commit message.

File: cardcomm/ctkToken/BEIDTokenApp/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 3]
```diff
@@ -1,63 +1,68 @@
 {
   "images" : [
     {
-      "size" : "16x16",
-      "idiom" : "mac",
       "filename" : "eID 16.png",
-      "scale" : "1x"
+      "idiom" : "mac",
+      "scale" : "1x",
+      "size" : "16x16"
```
Comment (yoe) [inline cardcomm/ctkToken/BEIDTokenApp/Assets.xcassets/AppIcon.appiconset/Contents.json:7]: This file just reorders members. That's unnecessary noise. Please don't add that, or put it in a separate commit if you really must.

Also, please don't commit the icon files together with code. They should be in separate commits.

File: cardcomm/pkcs11/src/cardlayer/card.cpp [hunk 4]
```diff
@@ -42,9 +42,9 @@ using namespace eIDMW;
 static const unsigned char APPLET_AID[] =
 { 0xA0, 0x00, 0x00, 0x00, 0x30, 0x29, 0x05, 0x70, 0x00, 0xAD, 0x13,
 0x10, 0x01, 0x01, 0xFF };
-static const unsigned char BELPIC_AID[] =
+/*static const unsigned char BELPIC_AID[] =
 { 0xA0, 0x00, 0x00, 0x01, 0x77, 0x50, 0x4B, 0x43, 0x53, 0x2D, 0x31,
-0x35 };
+0x35 };*/
```
Comment (yoe) [inline cardcomm/pkcs11/src/cardlayer/card.cpp:47]: ?

No.

Why on earth would we even want to drop support for a whole generation of ID cards?

File: cardcomm/pkcs11/src/cardlayer/card.cpp [hunk 5]
```diff
@@ -587,7 +587,7 @@ namespace eIDMW
 
 				if (ulDelay != 0)
 				{
-					CThread::SleepMillisecs(ulDelay);
+					CThread::SleepMillisecs((int)ulDelay);
```
Comment (yoe) [inline cardcomm/pkcs11/src/cardlayer/card.cpp:590]: Please explain in the commit message what problem this kind of change solves.

File: cardcomm/pkcs11/src/common/dynamiclib.cpp [hunk 6]
```diff
@@ -122,34 +122,48 @@ void CDynamicLib::PlatformClose()
 #include <Carbon/Carbon.h>
 #include <mach-o/dyld.h>
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1050
+#include <dlfcn.h>
+#endif
+
```
Comment (yoe) [inline cardcomm/pkcs11/src/common/dynamiclib.cpp:128]: Please put this change in a separate commit.

File: cardcomm/pkcs11/src/common/libtomcrypt/md5.c [hunk 7]
```diff
@@ -104,9 +104,16 @@ static int _md5_compress(hash_state *md, unsigned char *buf)
 static int  md5_compress(hash_state *md, unsigned char *buf)
 #endif
 {
+#ifdef __LP64__
+    ulong64 i, W[16], a, b, c, d;
+#ifdef LTC_SMALL_CODE
```
Comment (yoe) [inline cardcomm/pkcs11/src/common/libtomcrypt/md5.c:109]: Please explain in the commit message what purpose this change serves.

File: cardcomm/pkcs11/src/common/libtomcrypt/sha1.c [hunk 8]
```diff
@@ -47,9 +47,9 @@ static int _sha1_compress(hash_state *md, unsigned char *buf)
 static int  sha1_compress(hash_state *md, unsigned char *buf)
 #endif
 {
-    ulong32 a,b,c,d,e,W[80],i;
+    unsigned long a,b,c,d,e,W[80],i;
```
Comment (yoe) [inline cardcomm/pkcs11/src/common/libtomcrypt/sha1.c:50]: No. "unsigned long" is not the same thing as a 32-bit integer. There is a strong reason why this needs to be a 32-bit integer, rather than just "unsigned long".

File: cardcomm/pkcs11/src/common/libtomcrypt/tomcrypt_hash.h [hunk 9]
```diff
@@ -29,49 +29,101 @@ extern "C"
...
+    ulong64 state[8];
+    unsigned long curlen;
+    unsigned char buf[128];
+};
+#else
+    struct sha512_state
 	{
-		ulong64 length, state[8];
+        ulong32 length;
+        ulong32 state[8];
```
Comment (yoe) [inline cardcomm/pkcs11/src/common/libtomcrypt/tomcrypt_hash.h:44]: No. You're dropping bits here, for a SHA512 algorithm. That's a terrible idea.

Did you run the hashing code, and did it return the same value?

File: cardcomm/pkcs11/src/common/libtomcrypt/tomcrypt_macros.h [hunk 10]
```diff
@@ -212,7 +212,7 @@ asm("roll %%cl,%0": "=r"(word):"0"(word), "c"(i));
 	return word;
 }
 
-static inline unsigned ROR(unsigned word, int i)
+static inline unsigned ROR(unsigned word, long i)
```
Comment (yoe) [inline cardcomm/pkcs11/src/common/libtomcrypt/tomcrypt_macros.h:215]: No. You can't just change the type of a parameter to an asm method without causing problems.

File: cardcomm/pkcs11/src/common/tlvbuffer.cpp [hunk 11]
```diff
@@ -429,11 +429,11 @@ Decode the length from the TLV format
    
  **********************************************************************************************************/
 	bool CTLVBuffer::TlvDecodeLen(const unsigned char *pucBufSrc,	//In: Source buffer with the encoded length-stream
-				      int *piBufLen,	//In/Out: Maximum length of encoded stream, Out:number of length-bytes
+				      long *piBufLen,	//In/Out: Maximum length of encoded stream, Out:number of length-bytes
```
Comment (yoe) [inline cardcomm/pkcs11/src/common/tlvbuffer.cpp:432]: Why?

File: cardcomm/pkcs11/src/common/tlvbuffer.cpp [hunk 12]
```diff
@@ -429,11 +429,11 @@ Decode the length from the TLV format
...
 				      unsigned long *pulLenVal	//Out: length-value from the TLV stream
 		)
 	{
-		int iStreamLenMax = 0;	//length of the stream
+		long iStreamLenMax = 0;	//length of the stream
```
Comment (yoe) [inline cardcomm/pkcs11/src/common/tlvbuffer.cpp:436]: Why?

File: cardcomm/pkcs11/src/common/tlvbuffer.cpp [hunk 13]
```diff
@@ -488,7 +498,7 @@ Decode the length from the TLV format
 
 		CByteArray oTempArray;
 		unsigned char ucLenBuffer[4] = { 0 };
-		int iBufLen = sizeof(ucLenBuffer);
+		long iBufLen = sizeof(ucLenBuffer);
```
Comment (yoe) [inline cardcomm/pkcs11/src/common/tlvbuffer.cpp:501]: If we're going to change the type here, it should be to size_t.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj [hunk 14]
```diff
@@ -118,19 +118,19 @@
...
 			attributes = {
-				LastUpgradeCheck = 0610;
+				LastUpgradeCheck = 1240;
 				ORGANIZATIONNAME = FedICT;
 				TargetAttributes = {
 					658054E41A66C9BB000AEFE0 = {
 						CreatedOnToolsVersion = 6.1.1;
-						DevelopmentTeam = D4SW6VZB8M;
-						ProvisioningStyle = Manual;
+						DevelopmentTeam = YZAQ856TTM;
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj:126]: No.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj [hunk 15]
```diff
@@ -222,18 +222,21 @@
...
 				CLANG_WARN_STRICT_PROTOTYPES = YES;
 				CLANG_WARN_SUSPICIOUS_MOVE = YES;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
-				CODE_SIGN_IDENTITY = "-";
+				CODE_SIGN_IDENTITY = "Apple Development";
 				CONFIGURATION_BUILD_DIR = "$(PROJECT_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)";
 				COPY_PHASE_STRIP = NO;
-				DEVELOPMENT_TEAM = D4SW6VZB8M;
+				DEVELOPMENT_TEAM = YZAQ856TTM;
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj:234]: No.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj [hunk 16]
```diff
@@ -246,7 +249,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				MACOSX_DEPLOYMENT_TARGET = 10.7;
+				MACOSX_DEPLOYMENT_TARGET = 11.1;
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj:252]: No.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj [hunk 17]
```diff
@@ -275,27 +278,30 @@
...
 				CLANG_WARN_SUSPICIOUS_MOVE = YES;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
-				CODE_SIGN_IDENTITY = "Developer ID Application";
+				CODE_SIGN_IDENTITY = "Apple Development";
 				CONFIGURATION_BUILD_DIR = "$(PROJECT_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)";
 				COPY_PHASE_STRIP = YES;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
-				DEVELOPMENT_TEAM = D4SW6VZB8M;
+				DEVELOPMENT_TEAM = YZAQ856TTM;
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj:291]: No.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj [hunk 18]
```diff
@@ -275,27 +278,30 @@
...
 				GCC_C_LANGUAGE_STANDARD = gnu99;
+				GCC_NO_COMMON_BLOCKS = YES;
 				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
 				GCC_WARN_ABOUT_RETURN_TYPE = YES_ERROR;
 				GCC_WARN_UNDECLARED_SELECTOR = YES;
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				MACOSX_DEPLOYMENT_TARGET = 10.7;
+				MACOSX_DEPLOYMENT_TARGET = 11.1;
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj:302]: No.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj [hunk 19]
```diff
@@ -275,27 +278,30 @@
...
 				MTL_ENABLE_DEBUG_INFO = NO;
+				OTHER_CODE_SIGN_FLAGS = "--timestamp";
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj:304]: Why? Probably no.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj [hunk 20]
```diff
@@ -305,15 +311,12 @@
...
 				ARCHS = "$(ARCHS_STANDARD)";
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CODE_SIGN_IDENTITY = "-";
-				CODE_SIGN_STYLE = Automatic;
 				COMBINE_HIDPI_IMAGES = YES;
-				DEVELOPMENT_TEAM = D4SW6VZB8M;
 				INFOPLIST_FILE = "eID Middleware/Info.plist";
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks";
-				MACOSX_DEPLOYMENT_TARGET = 10.7;
+				MACOSX_DEPLOYMENT_TARGET = 11.1;
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj:317]: No.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj [hunk 21]
```diff
@@ -322,15 +325,12 @@
...
 				ARCHS = "$(ARCHS_STANDARD)";
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CODE_SIGN_IDENTITY = "Mac Developer";
-				CODE_SIGN_STYLE = Automatic;
 				COMBINE_HIDPI_IMAGES = YES;
-				DEVELOPMENT_TEAM = D4SW6VZB8M;
 				INFOPLIST_FILE = "eID Middleware/Info.plist";
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks";
-				MACOSX_DEPLOYMENT_TARGET = 10.7;
+				MACOSX_DEPLOYMENT_TARGET = 11.1;
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware.xcodeproj/project.pbxproj:331]: No.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware/AppDelegate.h [hunk 22]
```diff
@@ -19,7 +19,10 @@
 
 **************************************************************************** */
 
-#import <Cocoa/Cocoa.h>
+#import <Foundation/Foundation.h>
+#import <AppKit/AppKit.h>
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware/AppDelegate.h:23]: What benefit does this give us? Please explain in the commit message.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware/Base.lproj/MainMenu.xib [hunk 23]
```diff
@@ -57,6 +57,23 @@
...
+                                </connections>
+                            </menuItem>
+                            <menuItem title="Print…" keyEquivalent="p" id="kcM-FJ-Kbh">
+                                <connections>
+                                    <action selector="print:" target="-1" id="RWV-kv-9P9"/>
+                                </connections>
+                            </menuItem>
+                        </items>
+                    </menu>
+                </menuItem>
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware/Base.lproj/MainMenu.xib:76]: Why? Please explain in the commit message what the changes in this file accomplish.

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware/DataItem.h [hunk 24]
```diff
@@ -20,10 +20,11 @@
 **************************************************************************** */
 
 #import <Foundation/Foundation.h>
-@ import AppKit;
+#import <AppKit/AppKit.h>
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware/DataItem.h:23]: Why?

File: plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware/Info.plist [hunk 25]
```diff
@@ -7,9 +7,11 @@
 	<key>CFBundleExecutable</key>
 	<string>$(EXECUTABLE_NAME)</string>
 	<key>CFBundleIconFile</key>
-	<string></string>
+	<string>AppIcon</string>
+	<key>CFBundleIconName</key>
+	<string>AppIcon</string>
 	<key>CFBundleIdentifier</key>
-	<string>be.fedict.eid.$(PRODUCT_NAME:rfc1034identifier)</string>
+	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
```
Comment (yoe) [inline plugins_tools/aboutmw/OSX/eID Middleware/eID Middleware/Info.plist:14]: Why?

File: plugins_tools/eid-viewer/OSX/eID Viewer/eID Viewer/CertificateStore.m [hunk 26]
```diff
@@ -190,6 +190,9 @@ -(void)outlineViewSelectionDidChange:(NSNotification *)notification {
 		case eIDResultUnknown:
 			trust = @"";
 			break;
+        case eIDResultWarning:
```
Comment (yoe) [inline plugins_tools/eid-viewer/OSX/eID Viewer/eID Viewer/CertificateStore.m:193]: Please indent consistently with other indented items at the same location.

File: plugins_tools/eid-viewer/OSX/eID Viewer/eID Viewer/oslayer-objc.m [hunk 27]
```diff
@@ -1,7 +1,7 @@
 #import "oslayer-objc.h"
 #include <state.h>
 #include <pthread.h>
-#include "verify_cert.h"
+#include <verify_cert.h>
```
Comment (yoe) [inline plugins_tools/eid-viewer/OSX/eID Viewer/eID Viewer/oslayer-objc.m:4]: Why? Please explain in the commit message.

Why do this for verify_cert.h but not for the other two include items?

File: plugins_tools/eid-viewer/OSX/support/ossl/GNUmakefile [hunk 28]
```diff
@@ -1,53 +1,131 @@
-OSSL_VERSION=1.1.1i
-TP_DIR=../../../../../../ThirdParty
+# GNU Makefile for building ReSSL
+#
+LIBRESSL_VERSION=3.3.1
+TP_DIR=$(shell pwd)
```
Comment (yoe) [inline plugins_tools/eid-viewer/OSX/support/ossl/GNUmakefile:4]: No.

This is outside of the git checkout directory for a reason. Please keep it that way.

**Review by yoe — CHANGES_REQUESTED** [review]
Just spent 45 minutes reviewing the first commit in this PR. Needless to say, it can't accept it a is. I got bored at the end, so more changes are probably required, but...

Please research and understand the concept of "atomic commits", before trying again. You don't *have* to use them, but something a *bit* closer to that concept will absolutely be required, otherwise we won't be able to properly review your proposed changes.

That's not to say that there aren't any proposed changes in this massive commit that I may want to accept. Just not the way things are right now.

Here's what I want to see next time:

"Fix problem XYZ

The file foo/bar/foobar.m tries to do XYZ, but using an API call that has been deprecated with macOS X.Y. Reimplement using the new API"

and then a changeset containing *only* the required changes to fix *that* particular deprecation.

It's fine if you group similar changes across multiple files into one commit. But a commit should *not* have an enumeration of four different types of changes -- that alone should be an indication that you need more than one commit.

**Comment by yoe** [comment]
Changes were requested, but these never happened.

Closing.
